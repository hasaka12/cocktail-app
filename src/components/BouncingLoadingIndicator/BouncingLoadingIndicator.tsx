export default function BouncingLoadingIndicator() {
  return (
    <div className="flex justify-center bouncing-loader gap-2">
      <div
        className="bg-blue-500 w-3 h-3 rounded-full animate-bounce"
        style={{ animationDelay: "0ms" }}
      ></div>
      <div
        className="bg-blue-500 w-3 h-3 rounded-full animate-bounce"
        style={{ animationDelay: "200ms" }}
      ></div>
      <div
        className="bg-blue-500 w-3 h-3 rounded-full animate-bounce"
        style={{ animationDelay: "500ms" }}
      ></div>
    </div>
  );
}
