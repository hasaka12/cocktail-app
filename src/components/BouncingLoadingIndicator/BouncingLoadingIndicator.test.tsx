import React from "react";
import { render } from "@testing-library/react";
import BouncingLoadingIndicator from "./BouncingLoadingIndicator";

describe("CocktailCard", () => {
  test("renders three bouncing loader elements", () => {
    const { container } = render(<BouncingLoadingIndicator />);
    const loaderElements = container.querySelectorAll(".bouncing-loader div");
    expect(loaderElements.length).toBe(3);
  });

  test("renders loader elements with correct animation delay", () => {
    const { container } = render(<BouncingLoadingIndicator />);
    const loaderElements = container.querySelectorAll(".bouncing-loader div");
    const animationDelays = Array.from(loaderElements).map(
      (element: any) => element.style.animationDelay
    );
    expect(animationDelays).toEqual(["0ms", "200ms", "500ms"]);
  });

  test("Renders correctly", () => {
    const { asFragment } = render(<BouncingLoadingIndicator />);
    expect(asFragment()).toMatchSnapshot();
  });
});
