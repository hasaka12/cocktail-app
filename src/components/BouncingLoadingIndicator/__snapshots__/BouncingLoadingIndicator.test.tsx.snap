// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CocktailCard Renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center bouncing-loader gap-2"
  >
    <div
      class="bg-blue-500 w-3 h-3 rounded-full animate-bounce"
      style="animation-delay: 0ms;"
    />
    <div
      class="bg-blue-500 w-3 h-3 rounded-full animate-bounce"
      style="animation-delay: 200ms;"
    />
    <div
      class="bg-blue-500 w-3 h-3 rounded-full animate-bounce"
      style="animation-delay: 500ms;"
    />
  </div>
</DocumentFragment>
`;
